src/main/java/ch/njol/skript/test/runner/ExprSecRunnable.java

```diff
@@ -0,0 +1,87 @@
+package ch.njol.skript.test.runner;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.NoDoc;
+import ch.njol.skript.expressions.base.SectionExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+@NoDoc
+public class ExprSecRunnable extends SectionExpression<Object> {
+
+	public static class RunnableEvent extends Event {
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprSecRunnable.class, Object.class, ExpressionType.SIMPLE, "[a] new runnable");
+	}
+
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] expressions,
+						int pattern,
+						Kleenean delayed,
+						ParseResult result,
+						@Nullable SectionNode node,
+						@Nullable List<TriggerItem> triggerItems) {
+		if (node == null) {
+			Skript.error("Runnable expression needs a section!");
+			return false;
+		}
+		this.trigger = this.loadCode(node, "runnable", null, RunnableEvent.class);
```

Comment:
I don't think this is a good use case for this method. You should be able to use the regular loadCode method, right? Maybe a method like `runSection(Event)` for simplicity.

Reply:
I'll be honest I don't really understand. This class is just a test to make sure that you can use whatever's inside the section in a meaningful way. If you're suggesting an extra method or something you'll have to tell me where to put it (or add it yourself).

Reply:
I suppose it doesn't matter since it is a test, but typically the method that returns a trigger should only be used when absolutely necessary as changing the current event prevents context from carrying over (and you have to copy over variables). I can see about simplifying it.